### PR TITLE
all: sample formatting with google-java-format

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
DO NOT SUBMIT!

cc: @zhangkun83 @ejona86 @nmittler 

This is a sample of running our code though the standard google formatter.  The  style is a little wonky in some places, but is consistent, and would be _the_ way to do formatting.   I have created this PR to see what you think.

### CODE REVIEW

File: core/src/main/java/io/grpc/DnsNameResolver.java [hunk 1]
```diff
@@ -85,8 +94,9 @@
     // Must prepend a "//" to the name when constructing a URI, otherwise it will be treated as an
     // opaque URI, thus the authority and host of the resulted URI would be null.
     URI nameUri = URI.create("//" + name);
-    authority = Preconditions.checkNotNull(nameUri.getAuthority(),
-        "nameUri (%s) doesn't have an authority", nameUri);
+    authority =
+        Preconditions.checkNotNull(
+            nameUri.getAuthority(), "nameUri (%s) doesn't have an authority", nameUri);
```
Comment (ejona86) [inline core/src/main/java/io/grpc/DnsNameResolver.java]: The extra newline for this LOC is unfortunate. (There are much worse cases, but you might claim you should simplify the code in those cases. This case is already pretty simple.)

**Comment by ejona86** [comment]
In general it seems to be making a lot of nice changes. I'm pretty close to welcoming our formatting czar. If it weren't for those multi-line assignments that become bloated there would be no reason not to.

**Comment by zhangkun83** [comment]
I think the pros of auto-formatting and a more consistent style are greater than the cons. Does it also remove unused imports?

**Comment by nmittler** [comment]
@carl-mastrangelo looks great! :)

**Comment by carl-mastrangelo** [comment]
@zhangkun83  It doesn't handle unused imports unfortunately.

@ejona86 The multiline wrapping is unfortunate.  We can work around that by making lines shorter (terser error messages, statically importing preconditions)  Obviously not perfect, but I'm willing to do what the formatter says if it will do the work for me.

**Comment by ejona86** [comment]
The only issue left that we would have to discuss is what to do about updates to the java formatter. Unfortunately the formatter is not as nice as Go's and throws away the previous formatting, so updates to the formatter will likely produce noise. Are we okay with that?

**Comment by zhangkun83** [comment]
I suppose the noise from future updates of the formattter would be much less than that from the initial application, thus it should be OK.

**Comment by carl-mastrangelo** [comment]
It looks like in general we like this change, but I would wait until we have a presubmit check and a way to automatically do the formatting from command line.  I am going to close this, and likely reference it once I figure out how to do those things.

**Pull request closed without merging**